src/System.Linq/src/System/Linq/Zip.cs

```diff
@@ -28,6 +28,33 @@ public static partial class Enumerable
             return ZipIterator(first, second, resultSelector);
         }
 
+        public static IEnumerable<(TFirst First, TSecond Second)> Zip<TFirst, TSecond>(this IEnumerable<TFirst> first, IEnumerable<TSecond> second)
+        {
+            if (first is null)
+            {
+                throw Error.ArgumentNull(nameof(first));
+            }
+
+            if (second is null)
```

Comment:
Nit: Is there any benefit to using `is` instead of `==` here?  It just looks strange having it be different from the other overload above it.

Reply:
`is null` never takes into account user-defined conversions, so it always issues the optimal IL (brfalse) for a null check. It doesn't matter in this case because the argument is an interface so there can be no interface conversions, but that's the beauty of `is null` -- if you always use it you never have to think about whether or not the type itself has optimized for a null check.

Reply:
Thanks; that's what I thought.  FWIW, either we should stick with `==` here, or we should change the other similar occurrences to use `is`.  Otherwise this is just an oddity of style off by itself.